jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/SqlDialects.java

```diff
@@ -127,15 +127,17 @@ public SqlDialect getDialectInstanceForNameWithContext(final String name, final
 
     private SqlDialect instantiateDialect(final String name, final Optional<Class<? extends SqlDialect>> foundDialect,
             final SqlDialectContext context) throws SqlDialectsRegistryException {
-        SqlDialect instance;
         try {
-            final Class<? extends SqlDialect> dialectClass = foundDialect.get();
-            instance = dialectClass.getConstructor(SqlDialectContext.class).newInstance(context);
+            if (foundDialect.isPresent()) {
+                final Class<? extends SqlDialect> dialectClass = foundDialect.get();
+                return dialectClass.getConstructor(SqlDialectContext.class).newInstance(context);
+            } else {
+                throw new InstantiationException();
```

Comment:
Say that no dialect with that name exists and give users a list of supported dialect names.

Reply:
done